user/db.go

```diff
@@ -73,161 +70,276 @@ func (s *Store) updateUser(ctx context.Context, user User) error {
 func (s *Store) getUser(ctx context.Context, user User) (User, error) {
 	u := User{}
 	err := s.db.GetContext(ctx, &u,
-		`SELECT user_id, username, nickname, first_name, last_name, email, last_login, salt, password, avatar, use_gravatar
+		`SELECT *
 		FROM people.users
-		WHERE (username = $1 AND username != '') OR (email = $2 AND email != '') OR user_id = $3
-		LIMIT 1;`, user.Username, user.Email, user.UserID)
+		WHERE (username = $1 AND username != '') OR (email = $2 AND email != '') OR (ldap_username = $3 AND ldap_username != '') OR user_id = $4
+		LIMIT 1;`, user.Username, user.Email, user.LDAPUsername, user.UserID)
 	if err != nil {
 		return u, fmt.Errorf("failed to get user from db: %w", err)
 	}
 	return u, nil
 }
 
-// getUsers will get a group of users
+// getUsers will get users
 func (s *Store) getUsers(ctx context.Context) ([]User, error) {
 	var u []User
 	err := s.db.SelectContext(ctx, &u,
-		`SELECT user_id, username, nickname, first_name, last_name, email, last_login, avatar, use_gravatar
+		`SELECT *
 		FROM people.users;`)
 	if err != nil {
 		return nil, err
 	}
 	return u, nil
 }
 
-// getUsersIDA will get a group of users by userId ASC
-func (s *Store) getUsersIDA(ctx context.Context) ([]User, error) {
+// getUsersSizePage will get users with page size
+func (s *Store) getUsersSizePage(ctx context.Context, size, page int) ([]User, error) {
+	var u []User
+	// SELECT u.*, COUNT(*) / $1 AS pages
+	err := s.db.SelectContext(ctx, &u,
+		`SELECT u.*
+		FROM people.users u
+-- 		GROUP BY u, user_id, username, university_username, email, first_name, last_name, nickname, login_type, password, salt, avatar, last_login, reset_pw, enabled, created_at, created_by, updated_at, updated_by, deleted_at, deleted_by, use_gravatar, ldap_username
+		LIMIT $1
+		OFFSET $2;`, size, size*(page-1))
+	if err != nil {
+		return nil, err
+	}
+	return u, nil
+}
+
+// getUsersSearch will get users search
+func (s *Store) getUsersSearch(ctx context.Context, search string) ([]User, error) {
 	var u []User
 	err := s.db.SelectContext(ctx, &u,
-		`SELECT user_id, username, nickname, first_name, last_name, email, last_login, avatar, use_gravatar
+		`SELECT *
 		FROM people.users
-		ORDER BY user_id ASC;`)
+		WHERE (CAST(user_id AS TEXT) LIKE '%' || $1 || '%')
+		   OR (username LIKE '%' || $1 || '%')
+		   OR (nickname LIKE '%' || $1 || '%')
+		   OR (first_name LIKE '%' || $1 || '%')
+		   or (last_name LIKE '%' || $1 || '%')
+		   OR (email LIKE '%' || $1 || '%')
+		   OR (first_name || ' ' || last_name LIKE '%' || $1 || '%');`, search)
 	if err != nil {
 		return nil, err
 	}
 	return u, nil
 }
 
-// getUsersIDD will get a group of users by userId DESC
-func (s *Store) getUsersIDD(ctx context.Context) ([]User, error) {
+// getUsersSearchSizePage will get users search with size and page
```

Comment:
Same as above, redundant comment, either add more explanation to the parameters' meanings or remove it entirely. (There's quite a few other comments like this in this file, so please look over them in lieu of me commenting on each one.)

Reply:
Also added comments to all